src/log.cr

```diff
@@ -66,28 +66,45 @@
 # end
 # ```
 #
+# ### Default logging configuration
+#
+# By default entries from all sources with `Info` and above severity will
+# be logged to `STDOUT` using the `Log::IOBackend`.
+#
+# If you need to change the default level, backend or sources call `Log.setup` upon startup.
+#
+# ```
+# Log.setup(:debug)                     # Log debug and above for all sources to STDOUT
+# Log.setup("myapp.*, http.*", :notice) # Log notice and above for myapp.* and http.* sources only, and log nothing for any other source.
+# backend_with_formatter = Log::IOBackend.new(formatter: custom_formatter)
+# Log.setup(:debug, backend_with_formatter) # Log debug and above for all sources to using a custom backend
+# ```
+#
 # ### Configure logging explicitly in the code
 #
-# Use `Log.builder` to indicate which sources should go to which backends.
+# Use `Log.setup` methods to indicate which sources should go to which backends.
 #
 # You can indicate actual sources or patterns.
 #
 # * the empty string matches only the top-level source
 # * `*` matches all the sources
 # * `foo.bar.*` matches `foo.bar` and every nested source
 # * `foo.bar` matches `foo.bar`, but not its nested sources
-#
+# * Any comma separated combination of the above
 #
 # The following configuration will setup for all sources to emit
 # warnings (or higher) to `STDOUT`, allow any of the `db.*` and
 # nested source to emit debug (or higher), and to also emit for all
 # sources errors (or higher) to an elasticsearch backend.
 #
 # ```
-# backend = Log::IOBackend.new
-# Log.builder.bind "*", :warning, backend
-# Log.builder.bind "db.*", :debug, backend
-# Log.builder.bind "*", :error, ElasticSearchBackend.new("http://localhost:9200")
+# Log.setup |c|
+#   backend = Log::IOBackend.new
+#
+#   c.bind "*", :warning, backend
+#   c.bind "db.*", :debug, backend
+#   c.bind "*", :error, ElasticSearchBackend.new("http://localhost:9200")
```

Comment:
I am not sure when to use `Log.setup` v `bind`. Maybe that could be explain here? It seems to me that `Log.setup` does the same thing but I'm not sure

Reply:
A `Log.setup` will configure entirely the logging module.
`c.bind` will be used to define the binding.

If you only need 1 bind, there is an overload of `Log.setup` that does that.

Since calling `Log.setup` clears the existing binding, the last call of `Log.setup` is the one that counts.

Reply:
Oh I see. I misunderstood what setup was doing